### PR TITLE
Azure ARM deployment instructions

### PR DESCRIPTION
**Pull request by sangramrath**
PR includes Azure ARM template to perform a somewhat one-click deployment on Azure. Instructions on executing the template along with some troubleshooting help is also included.

### CODE REVIEW

**Review by SebastianStehle — COMMENTED** [review]
Looks good, but I have issues to understand how to make any changes to the deployment.

File: scripts/squidex-minimal-azure-arm.json [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+      "defaultValue": "squidex-crm",
+      "type": "String",
+      "metadata": {
+        "description": "The name of the Azure Container Instance Group"
+      }
+    }
+  },
+  "variables": {
+    "dnsNameLabel": "[concat('squidex',uniqueString(resourceGroup().id))]",
+    "mongodbContainerName": "[concat(variables('storageContainerName'), '-mongodb')]",
```
[the remark this answers is not in the document]
Comment (sangramrath) [reply scripts/squidex-minimal-azure-arm.json]: Yes, corrected now.

File: 01-getting-started/installation/platforms/install-on-azure-arm.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+7. Click **Create** to submit the deployment.
+
+![Review ARM Template](../../../images/started/azure/squidex-arm-02.png)
+
+The deployment process will take a few minutes.
+
+## 3. Access Squidex
+
+1. Click **Outputs** (8) in the deployment page to retrieve the URL.
+2. Copy the URL (9).
```
[the remark this answers is not in the document]
Comment (sangramrath) [reply 01-getting-started/installation/platforms/install-on-azure-arm.md:77]: Added instructions to fetch it from ACI.

File: 01-getting-started/installation/platforms/install-on-azure-arm.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  * Default resource allocations
+    * Squidex container: **1 vCPU and 1.5 GiB memory**
+    * MongoDB container: **2 vCPU and 2 GiB memory**
+    * Caddy container: **1 vCPU and 1.5 GiB memory**
+  * Squidex and MongoDB containers remain private and are not exposed to public.
+  * Squidex talks to MongoDB without authentication (mongodb://localhost:27017). Since MongoDB is private it cannot be accessed from outside, hence secure to an extent. **With the current setup you cannot connect using an external client like MongoDB Compass. You can however connect from Azure Container Instance, click [here](#connecting-to-mongodb-instance) for instructions**
+  * Caddy reverse proxies to the Squidex Web App container.
+  * An FQDN is provided by Azure Container Instance, _Caddy_ provides SSL for this as the FQDN does not come with SSL. This FQDN is set as the `URLS__BASEURL` and `VIRTUAL_HOST` for Squidex configuration.
+
+**NOTE: A future version of this template will support bringing your own domain and other customizations.**
```
[the remark this answers is not in the document]
Comment (sangramrath) [reply 01-getting-started/installation/platforms/install-on-azure-arm.md:46]: I figured out a solution for this. I will update code and instructions shortly.

[the remark this answers is not in the document]
Comment (sangramrath) [reply 01-getting-started/installation/platforms/install-on-azure-arm.md]: Updated script and instructions.

**Comment by SebastianStehle** [comment]
Now it looks like this: 
image

File: 01-getting-started/installation/platforms/install-on-azure-arm.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  * Default resource allocations
+    * Squidex container: **1 vCPU and 1.5 GiB memory**
+    * MongoDB container: **2 vCPU and 2 GiB memory**
+    * Caddy container: **1 vCPU and 1.5 GiB memory**
+  * Squidex and MongoDB containers remain private and are not exposed to public.
+  * Squidex talks to MongoDB without authentication (mongodb://localhost:27017). Since MongoDB is private it cannot be accessed from outside, hence secure to an extent. **With the current setup you cannot connect using an external client like MongoDB Compass. You can however connect from Azure Container Instance, click [here](#connecting-to-mongodb-instance) for instructions**
+  * Caddy reverse proxies to the Squidex Web App container.
+  * An FQDN is provided by Azure Container Instance, _Caddy_ provides SSL for this as the FQDN does not come with SSL. This FQDN is set as the `URLS__BASEURL` and `VIRTUAL_HOST` for Squidex configuration.
+
+**NOTE: If you wish to modify any of the variables such as vCPU, Memory etc., you can do so by editing the ARM template. Click [here](#modifying-arm-variables) for basic instructions.**
```
Comment (SebastianStehle) [inline 01-getting-started/installation/platforms/install-on-azure-arm.md:46]: For something like the notes here, you can use hints

```
{% hint style="info" %} YOUR HINT  {% endhint %
```

See: https://docs.squidex.io/01-getting-started/contributing-and-developing/developing

**Comment by sangramrath** [comment]
> Now it looks like this: image

Yes, that's correct. Leaving it as it is will mean using ACI DNS, user replaces with their own domain if needed. This way both are possible.

File: 01-getting-started/installation/platforms/install-on-azure-arm.md (hunk 4 above)
Comment (sangramrath) [reply 01-getting-started/installation/platforms/install-on-azure-arm.md]: Done

**Comment by SebastianStehle** [comment]
So it is not possible to set a placeholder here?

**Comment by SebastianStehle** [comment]
Some variable like

```
"actualDomain": "[if(empty(parameters('squidexDomain')), [concat('squidex', uniqueString(resourceGroup().id), '.', resourceGroup().location, '.azurecontainer.io')], parameters('squidexDomain'))]
```

**Comment by sangramrath** [comment]
> So it is not possible to set a placeholder here?

Meaning, the field should be blank?

**Comment by SebastianStehle** [comment]
I think this would be best, otherwise you can easily make mistakes, for example when you add a character by accident to your concat logic.

**Comment by sangramrath** [comment]
I made changes to the template, you can test it again now.

File: scripts/squidex-minimal-azure-arm.json [hunk 5]
```diff
@@ -0,0 +1,277 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "containerName": {
+      "defaultValue": "squidex",
+      "type": "String",
+      "metadata": {
+        "description": "The name of the Azure Container Instance Group"
```
Comment (SebastianStehle) [inline scripts/squidex-minimal-azure-arm.json:9]: Can you change this to "Name of your deployment and prefix for all container names."?

File: scripts/squidex-minimal-azure-arm.json [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+      }
+    },
+    "squidexDomain": {
+      "defaultValue": "",
+      "type": "String",
+      "metadata": {
+        "description": "The Domain/Sub-domain to be used in Squidex."
```
Comment (SebastianStehle) [inline scripts/squidex-minimal-azure-arm.json:16]: Can you change this "Optional domain name (FQDN) if you want to host Squidex under a custom domain. If not defined, a domain name will be assigned."

File: scripts/squidex-minimal-azure-arm.json (hunk 5 above)
Comment (sangramrath) [reply scripts/squidex-minimal-azure-arm.json]: Done

File: scripts/squidex-minimal-azure-arm.json (hunk 6 above)
Comment (sangramrath) [reply scripts/squidex-minimal-azure-arm.json]: Done

**Pull request merged**